File: Javascript/ValidateBinarySearchTree.js
function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0:val
    this.left = (left === undefined) ? null:left
    this.right = (right === undefined) ? null:right
}

var isBST = function(root){

    function helper(node,left,right){
        if (node === null){
            return true
        }

        if(!(node.val>left && node.val<right)){
            return false
        }

        return (helper(node.left,left,node.val) && helper(node.right,node.val,right))
    }

    return helper(root,Number.NEGATIVE_INFINITY,Number.POSITIVE_INFINITY)

}

File: Python/ValidateBinarySearchTree.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def isBST(root):

        def helper(node,left,right):
            if not node:
                return True
            
            if not (node.val>left and node.val<right):
                return False
            
            return (helper(node.left,left,node.val) and helper(node.right,node.val,right))

        return helper(root,float("-inf"),float("inf"))